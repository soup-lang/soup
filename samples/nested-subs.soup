func even(x)
    return x % 2 = 0

if even(0)
    x: 42
    try
        console.log(x)
    catch e
        console.log(e)

if even(2)
    x: 42
    setTimeout(%, 0)
    try
        x: even(1)
        console.log(x)
    catch e
        console.log(e)

try
    setTimeout(%, 0)
    x: 42
    if even(4)
        x: even(3)
        console.log(x)
catch e
    console.log(e)

console.log('end')
