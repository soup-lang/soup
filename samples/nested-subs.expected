42
false
false
end
